pkg/service/helper.go

```diff
@@ -32,3 +32,22 @@ func diskAttachmentByVmAndDisk(connection *ovirtsdk.Connection, vmId string, dis
 		diskId: diskId,
 	}
 }
+
+func diskAttachmentByDisk(connection *ovirtsdk.Connection, diskId string) (*ovirtsdk.DiskAttachment, *ovirtsdk.DiskAttachmentService, error) {
+	vmsService := connection.SystemService().VmsService()
```

Comment:
Not sure I like going over all the VMs to get the disk attachment, I haven't tested it yet but I think something like this should work:
```
diskResp := connection.SystemService().DisksService().DiskService(diskId).Get().Send()
...error handling...
disk, err := diskResp.Disk()
...error handling...
diskVm, err := disk.Vm()
...error handling...
diskVmDiskAttachments, err := diskVm.DiskAttachments()
...error handling...
for _, attachment := range diskVmDiskAttachments.Slice() {
    if attachment.MustId() == diskId {
        return attachment, nil
    }
}
```

@bennyz what do you think? am I missing something? also if the above is ok should we use:
```
diskVms, err := disk.Vms()
```
and go over all of them instead? why there is a disk.Vm and disk.Vms?

Reply:
sounds right to me, we should use `disk.Vms()`

> and go over all of them instead? why there is a disk.Vm and disk.Vms?
A shared disk can be attached to multiple VMs, we don't support this yet, but we should (to enable RWX PVs)

Reply:
> I think we support ONLINE VolumeExpansion right?

Yes the online expansion works!

Reply:
Maybe it is my mistake but i was not able to get attachments that way.

Reply:
you're right, we don't have disk -> vm mapping, it has to go through the VM, we could run the ovirt disk update in the node, that we we'd already have the VM, but I'm not sure if it would behave correctly in a shareable disk scenario

Reply:
Yet another case of the go-ovirt SDK providing useless fields :/

so @bennyz are you ok with the approach of the function? to go over all the VMs and check each disk attachment?
I just think it can lead to a lot of calls on big RHV envs.... I guess we can filter the VMs by tag like we do on the installer destroy code:
https://github.com/openshift/installer/blob/3699e08dd2743b332f6dc372207e637712aa1401/pkg/destroy/ovirt/destroyer.go#L57-L59

That way we will at least have fewer VMs to go over

Reply:
yes, if we have a tag we should filter to only search the cluster nodes